pkgs/applications/networking/instant-messengers/riot/riot-desktop.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenv, fetchFromGitHub, yarn2nix, makeWrapper, makeDesktopItem, electron, riot-web }:
+
+let
+  executableName = "riot-desktop";
+  version = "1.0.4";
+  riot-web-src = fetchFromGitHub {
+    owner = "vector-im";
+    repo = "riot-web";
+    rev = "v${version}";
+    sha256 = "152mi81miams5a7l9rd12bnf6wkd1r0lyicgr35r5fq0p6z7a4dk";
+  };
+
+in yarn2nix.mkYarnPackage rec {
+  name = "riot-desktop-${version}";
+  inherit version;
+
+  src = "${riot-web-src}/electron_app";
```

Comment:
I think you should be able to just make `riot-web-src` `src` here and do

```
sourceRoot = "electron_app";
```

since that's really the only reason you needed to do this.